raiden-ts/src/raiden.ts

```diff
@@ -925,13 +925,15 @@ export class Raiden {
    * </ol>
    *
    * @param amount - The amount to deposit on behalf of the target/beneficiary.
+   * @param depositing - Called before the deposit function is called
    * @param options - tx options
    * @param options.subkey - By default, if using subkey, main account is used to send transactions
    *    Set this to true if one wants to force sending the transaction with the subkey
    * @returns transaction hash
    */
   public async depositToUDC(
     amount: BigNumberish,
+    depositing?: () => void,
```

Comment:
I think having a callback for a specific step is not a good idea, this is going to be too specific and require multiple callbacks for multi-step methods. I'd say it should instead be like `onChange?: (event { type: 'APPROVED', payload: { txHash: string } }) => void` or some other type-safe and well-described event-like callback.

Reply:
I just did it because it was the simplest thing to do and our discussion on how to approach that is still ongoing but the point is valid indeed. Thanks for the suggestion it sounds really good.

Reply:
@andrevmatos could you have a look to the newest implementation